php-checks/src/test/resources/checks/NestedTernaryOperatorsCheck.php

```diff
@@ -0,0 +1,17 @@
+<?php
+  function func($gender, $is_married) {
+    $gender == "MALE" ? "Mr. " : function () use($is_married) { return $is_married ? "Mrs. " : "Miss "; };
+
+
+    $gender == "MALE" ? "Mr. " : ($is_married ? "Mrs. " : "Miss ");  // Noncompliant
+    //                ^>          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    $gender == "MALE" ?
+    //                ^> {{Parent ternary operator}}
+       ($is_married ? "Mrs. " : "Mrs. ") : ($is_married ? "Mrs. " : "Miss ");  // Noncompliant
+    //  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ {{Extract this nested ternary operation into an independent statement.}}
+
+    $is_married ? "Mrs. " : "Miss ";
+    $is_married ? : "Miss ";
+
```

Comment:
There is a (theoritical) case not handled in case of anonymous class : 
```
$cond ? 
new class {
  private $a = $foo ? "" : "bar";
  function someMethod() {}
}.someMethod() : $falseExpr
```